### ISSUE TITLE
A duplicate transaction should display "Duplicate Transaction ID" in the transaction description

### ISSUE THREAD

**Issue by SimiHunjan** [hashgraph/hedera-transaction-tool-demo#87]
### Problem

When two transactions with the same transaction ID are available to sign the application does not notify the user that the second transaction is a duplicate of another transaction. 

6:49

### Solution

Duplicate transactions should display "Duplicate Transaction ID" in the transaction description. 

If 2 or more transactions have the same transaction ID, then all but one of them should say “duplicate transaction ID” in the description.  And one of them should NOT be marked as a duplicate. If at least one of them has already been signed, then that's the one that should not be marked.

**Comment by DSMatusevich** [comment]
David Matusevich  10:12 AM
Regarding the duplicates feature in your first message, where would this be indicated? In the Home pane? Would a duplicate mean just the transaction id or, having identified transactions with the same id, the whole transaction? If the transactions are identical, it would be easy just to mark one as the real one and the rest as duplicates, however if they are different, I think an error should be thrown

Leemon  10:22 AM
The rectangle showing one transaction has many pieces of information on it. It could sometimes have one additional thing: a string saying “duplicate transaction”.  So when you see it, you know it is a duplicate.

I was only suggesting doing that. But I like your suggestion of checking whether two transitions with the same transaction ID have a field which is different (other than the submission node). In that case, you could add a red string saying “`ERROR: not all fields match the duplicate transaction`”.

David Matusevich  10:30 AM
I think I understand: When you are making transfers, you prepare several transactions with different nodes so that, in case one node is down, the other transactions will succeed, similarly to what the batch transactions do.
Wouldn’t it be better if this was done in the background? That the app prepares `n` transactions to the nodes, but shows only one and all duplicates are signed at the same time. That way there is less clutter for the user in the home pane and we can be sure that all transactions are the same.
In parallel we can add another feature that checks if there are transactions with the same TXID and flag them as `ERROR: Duplicate`.
10:31
So during transaction creation, instead of a single node, we can allow for a list of nodes and all the transactions would then be created identically and bundled together, with less opportunity for user error

Leemon  10:48 AM
It would be fine to merge them into a single rectangle, where the submission node is a list.

But then we have to handle the case of a few duplicates being signed, when some new duplicates are downloaded. Now we have several duplicates with different numbers of signatures on each one.
I had originally thought of merging them, but didn't see an easy way to handle this issue. So I just suggested putting the duplicate marker on them.
10:49
Maybe the best solution is to merge groups of them with the same set of signatures. And if there are multiple such groups, display them separately, with a duplicate marker on all but one.

Though that seems complicated. Maybe it's better to just go with the original suggestion.

David Matusevich  10:53 AM
Why would duplicates be created after the fact? My suggestion is to allow the user (transaction creator) to create them all in one go, and put them all together for the users to sign. We could do it as a zip of all transactions or similarly to what we do for the batch transactions, where the signers create and sign at the same time. This way there is no way that some transactions would have different number of signatures, since they would be created and signed at the same time.

Leemon  11:44 AM
We have to decide how to handle the case of someone not zipping it. We also need to handle the case of someone accidentally creating multiple transactions with the same transaction ID.
It would be great to say that if there are a number of duplicates inside a zip file, then they display as a single rectangle, and we ensure that we either sign all or none. If the computer crashes halfway through the signing, we would need to make sure that on restart, it recognizes that it only signed half the zip, and discards the old signatures and lets you try again from scratch.
But it would still need the alerts for additional duplicates that happen outside the zip files. We can't guarantee that the user will always remember to zip it. And we can't guarantee that OneDrive will download all the fils at once. So w still need the feature that I described originally.  In addition to this cool zip feature.
So I think that means that w now have 3 features to add:

1. If many duplicates are inside a zip file, show just one rectangle, and discard any partial signatures on startup (i.e., signatures on some of the duplicates but not on others).
2. If two transactions outside of zips are duplicates, then display them separately, but mark all but one as a duplicate. If one was already signed, then choose the signed one as the one to not mark.
3. If two duplicates (either inside a zip or outside it) have the same transaction ID but differ in any other field (other than submission node), then show a bright red warning "`ERROR: duplicate transaction ID` for transactions that contain different information".
